### PR TITLE
Remove torchgeo.datamodules.dataset_split

### PR DESCRIPTION
**Pull request by adamjstewart**
This function was added as a wrapper around `torch.utils.data.random_split` with support for percentages. However, this feature was added upstream, and the minimum supported version of PyTorch now supports it. This wrapper can now be removed.

There are no backwards incompatible changes since this function was never documented.

### CODE REVIEW

**Comment by adamjstewart** [comment]
Could be made simpler by allowing the user to input `lengths` directly, so they can choose how to define the split (percentages, numbers, etc.). Wouldn't be backwards compatible, but would also resolve #1289. However, many GeoDataModules already accept a `length` parameter, so I'm not sure what to call this. Maybe `split_lengths`?

File: torchgeo/datamodules/deepglobelandcover.py [hunk 1]
```diff
@@ -59,8 +60,9 @@ def setup(self, stage: str) -> None:
         """
         if stage in ["fit", "validate"]:
             self.dataset = DeepGlobeLandCover(split="train", **self.kwargs)
-            self.train_dataset, self.val_dataset = dataset_split(
-                self.dataset, self.val_split_pct
+            generator = torch.Generator().manual_seed(0)
```
Comment (isaaccorley) [inline torchgeo/datamodules/deepglobelandcover.py:63]: Should we be setting the random seed manually? Seems like this would conflict if a user is already setting a global random seed on purpose to evaluate random runs

Comment (adamjstewart) [reply torchgeo/datamodules/deepglobelandcover.py:63]: In the case a user runs `torchgeo fit` followed by `torchgeo validate`, we need to ensure that the same split is chosen both times. This is even more important for datamodules like NASA Marine Debris that choose train/val/test all at once. That's not to say that we can't or shouldn't add a `seed` parameter to all datamodules. I would be open to a PR that does this (so far this PR just preserves the current behavior).

Comment (isaaccorley) [reply torchgeo/datamodules/deepglobelandcover.py:63]: Ahh that's right we were already setting the seed previously. Carry on

**Comment by adamjstewart** [comment]
Let's merge this as is. We can think about renaming (lengths) or adding (random seed) parameters in a separate PR.
